### PR TITLE
fac: 2.0.0 -> 2.0.0-unstable-2023-12-29

### PR DESCRIPTION
**Pull request by katexochen**
## Description of changes

Last release was 2018, let's grab some minor fixes from main and the recent move to Go modules.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/by-name/fa/fac/package.nix [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+  vendorHash = "sha256-bmGRVTjleAFS5GGf2i/zN8k3SBtaEc3RbKSVZyF6eN4=";
+
+  nativeBuildInputs = [ makeWrapper ];
+
+  postInstall = ''
+    wrapProgram $out/bin/fac \
+      --prefix PATH : ${git}/bin
+
+    # Install man page, not installed by default
+    install -D assets/doc/fac.1 $out/share/man/man1/fac.1
```
Comment (SuperSandro2000) [inline pkgs/by-name/fa/fac/package.nix:28]:
```suggestion
    installManPage assets/doc/fac.1
```

we have a convince wrapper for installing man pages

File: pkgs/by-name/fa/fac/package.nix [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+    hash = "sha256-puSHbrzxTUebK1qRdWh71jY/f7TKgONS45T7PcZcy00=";
+  };
+
+  vendorHash = "sha256-bmGRVTjleAFS5GGf2i/zN8k3SBtaEc3RbKSVZyF6eN4=";
+
+  nativeBuildInputs = [ makeWrapper ];
+
+  postInstall = ''
+    wrapProgram $out/bin/fac \
+      --prefix PATH : ${git}/bin
```
Comment (SuperSandro2000) [inline pkgs/by-name/fa/fac/package.nix:25]:
```suggestion
      --prefix PATH : ${lib.makeBinPath [ git ]}
```
we don't need to construct that manually

**Comment by katexochen** [comment]
I've also updated the version with a prefix for the previous release, following

https://github.com/NixOS/nixpkgs/blob/e7069e4aa2b66cdc57a3f6812f96cb970081d63e/pkgs/README.md?plain=1#L353-L355

File: pkgs/by-name/fa/fac/package.nix [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+    installManPage assets/doc/fac.1
+  '';
+
+  meta = {
+    description = "CUI for fixing git conflicts";
+    homepage = "https://github.com/mkchoi212/fac";
+    changelog = "https://github.com/mkchoi212/fac/releases/tag/v${version}";
+    license = lib.licenses.mit;
+    maintainers = with lib.maintainers; [ dtzWill ];
+  };
```
Comment (drupol) [inline pkgs/by-name/fa/fac/package.nix:41]: Missing `meta.mainProgram`

File: pkgs/by-name/fa/fac/package.nix [hunk 4]
```diff
@@ -0,0 +1,42 @@
+{ lib
+, buildGoModule
+, fetchFromGitHub
+, makeWrapper
```
Comment (drupol) [inline pkgs/by-name/fa/fac/package.nix:4]: Prefer `makeBinaryWrapper` to `makeWrapper` for a better compatibility with Darwin architectures.

**Review by drupol — CHANGES_REQUESTED** [review]

**Review by drupol — APPROVED** [review]

**Pull request merged**